src/module.c

```diff
@@ -205,11 +206,18 @@ static void mvbox_thread_func(void* arg)
   napi_acquire_threadsafe_function(dcn_context->threadsafe_event_handler);
 #endif
 
+  printf("---> mvbox_thread_func\n");
+
   while (dcn_context->loop_thread) {
+    printf("---- before dc_perform_mvbox_fetch\n");
     dc_perform_mvbox_fetch(dc_context);
+    printf("---- before dc_perform_mvbox_idle\n");
```

Comment:
This is printed, so we know that `dc_perform_mvbox_idle()` is called below

Reply:
These before-printfs are also printed before the ones below, so we know this is happening before we try to close. Which leads me to believe that it's something failing when trying to interrupt `dc_perform_mvbox_idle()`